assay/src/org/labkey/assay/plate/AssayPlateMetadataServiceImpl.java

```diff
@@ -1045,6 +1054,64 @@ private void insertOrUpdateReplicateStats(
         }
     }
 
+    @Nullable
+    private static DataState getStateFromRow(Container container, Map<String, Object> row, @Nullable DomainProperty stateProp) throws ExperimentException
+    {
+        if (stateProp != null)
+        {
+            Set<String> importAlias = new CaseInsensitiveHashSet(stateProp.getName());
+            importAlias.addAll(stateProp.getImportAliasSet());
+            for (Map.Entry<String, Object> entry : row.entrySet())
+            {
+                if (importAlias.contains(entry.getKey()))
+                {
+                    if (entry.getValue() instanceof Integer stateRowId)
+                    {
+                        DataState state = PlateDataStateManager.get().getStateForRowId(container, stateRowId);
+                        if (state == null)
+                            throw new ExperimentException(String.format("No data states for the rowID : %d was found", stateRowId));
```

Comment:
as mentioned in the other comment, I'd suggest removing the colon from the error message and adding a period.
```suggestion
                            throw new ExperimentException(String.format("No data states for the rowID %d was found.", stateRowId));
```